cloudrail/knowledge/context/azure/azure_relations_assigner.py

```
@@ -171,3 +180,18 @@ def _assign_monitor_activity_log_alert_to_subscription(subscription: AzureSubscr
         subscription.monitor_activity_alert_log_list = ResourceInvalidator.get_by_logic(
             lambda: [monitor for monitor in monitor_activity_log_alert if subscription.get_id() in monitor.scopes],
             False)
+
+    @staticmethod
+    def _assign_network_rule_set_to_event_hub_namespace(network_rule_set: EventHubNetworkRuleSet, event_hub_namespaces: AliasesDict[AzureEventHubNamespace]):
+        event_hub: AzureEventHubNamespace = ResourceInvalidator.get_by_id(event_hub_namespaces,
+                                                                          network_rule_set.event_hub_namespace_id,
+                                                                          True, network_rule_set)
+        event_hub.network_rule_set = network_rule_set
```

Should we put here some validation that we have "event_hub" before trying to access it?

---

invalidator should throw exception if not found